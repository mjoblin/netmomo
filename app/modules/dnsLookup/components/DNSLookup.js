import moment from 'moment';
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Button, Dropdown, Icon, Input, Menu, Table } from 'antd';

import * as actionCreators from '../actions.js';
import { getAsArray, getHostComponentLevels } from '../selectors';
import './style.scss';


class DNSLookup extends React.Component {
    handlehostLevelSelection(e) {
        this.props.actions.setHostComponentLevels(Number(e.key));
    }

    handleHostFilterEntry(e) {
        this.props.actions.setHostFilter(e.target.value);
    }

    render() {
        const columns = [
            {
                title: 'Host',
                dataIndex: 'host',
                sorter: (a, b) => a.host.localeCompare(b.host),
            },
            {
                title: 'Count',
                dataIndex: 'count',
                sorter: (a, b) => a.count - b.count,
                className: 'column-count',
            },
            {
                title: 'Latest',
                dataIndex: 'latest',
                sorter: (a, b) => a.latest - b.latest,
                render: text =>
                    moment(text).format('MMMM DD YYYY, h:mm:ss a'),
            },
        ];

        const hostLevelsMenu = (
            <Menu onClick={e => this.handlehostLevelSelection(e)}>
                <Menu.Item key="0">All</Menu.Item>
                <Menu.Item key="1">1</Menu.Item>
                <Menu.Item key="2">2</Menu.Item>
                <Menu.Item key="3">3</Menu.Item>
                <Menu.Item key="4">4</Menu.Item>
            </Menu>
        );

        return (
            <div className="dns-lookup">
                <h2>DNS lookups</h2>

                <span className="label">{"Host levels: "}</span>
                <Dropdown overlay={hostLevelsMenu} trigger={['click']}>
                    <Button className="value">
                        {this.props.hostComponentLevels === 0 ?
                            'All' : this.props.hostComponentLevels}
                        <Icon type="down"/>
                    </Button>
                </Dropdown>

                <span className="label">Host filter:</span>
                <Input
                    className="value value-host-filter"
                    placeholder="Host filter"
                    onChange={e => this.handleHostFilterEntry(e)}
                />

                <Table
                    className="results-table"
                    dataSource={this.props.dnsLookups}
                    columns={columns}
                    pagination={false}
                    size="middle"
                />
            </div>
        );
    }
}


DNSLookup.propTypes = {
    actions: PropTypes.array.isRequired,
    dnsLookups: PropTypes.array.isRequired,
    hostComponentLevels: PropTypes.number.isRequired,
};

const mapStateToProps = state => ({
    dnsLookups: getAsArray(state),
    hostComponentLevels: getHostComponentLevels(state),
});

const mapDispatchToProps = dispatch => {
    return { actions: bindActionCreators(actionCreators, dispatch) };
};


export default connect(
    mapStateToProps,
    mapDispatchToProps
)(DNSLookup);
