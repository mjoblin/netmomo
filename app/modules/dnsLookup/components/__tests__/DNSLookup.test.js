import React from 'react';
import { shallow } from 'enzyme';
import { DNSLookup } from '../DNSLookup';

const testDNSLookups = [
    {
        key: 'two.com',
        host: 'two.com',
        count: 10,
        latest: 1509758701947.23,
    },
    {
        key: 'three.two.com',
        host: 'three.two.com',
        count: 3,
        latest: 1509758701932.883,
    },
    {
        key: 'four.three.two.com',
        host: 'four.three.two.com',
        count: 1,
        latest: 1509758701941.5847,
    },
];

describe('DNSLookup', () => {
    it('matches snapshot', () => {
        const component = shallow(
            <DNSLookup
                dnsLookups={testDNSLookups}
                hostComponentLevels={0}
                actions={{}}
            />
        );

        expect(component).toMatchSnapshot();
    });
});