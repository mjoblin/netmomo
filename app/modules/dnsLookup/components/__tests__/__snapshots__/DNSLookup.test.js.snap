// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DNSLookup matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DNSLookup
    actions={Object {}}
    dnsLookups={
      Array [
        Object {
          "count": 10,
          "host": "two.com",
          "key": "two.com",
          "latest": 1509758701947.23,
        },
        Object {
          "count": 3,
          "host": "three.two.com",
          "key": "three.two.com",
          "latest": 1509758701932.883,
        },
        Object {
          "count": 1,
          "host": "four.three.two.com",
          "key": "four.three.two.com",
          "latest": 1509758701941.5847,
        },
      ]
    }
    hostComponentLevels={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          DNS lookups
        </h2>,
        <span
          className="label"
        >
          Host levels: 
        </span>,
        <Dropdown
          mouseEnterDelay={0.15}
          mouseLeaveDelay={0.1}
          overlay={
            <Menu
              className=""
              onClick={[Function]}
              prefixCls="ant-menu"
              theme="light"
            >
              <MenuItem
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onSelect={[Function]}
              >
                All
              </MenuItem>
              <MenuItem
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onSelect={[Function]}
              >
                1
              </MenuItem>
              <MenuItem
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onSelect={[Function]}
              >
                2
              </MenuItem>
              <MenuItem
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onSelect={[Function]}
              >
                3
              </MenuItem>
              <MenuItem
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onSelect={[Function]}
              >
                4
              </MenuItem>
            </Menu>
          }
          placement="bottomLeft"
          prefixCls="ant-dropdown"
          trigger={
            Array [
              "click",
            ]
          }
        >
          <Button
            className="value"
            clicked={false}
            ghost={false}
            loading={false}
            prefixCls="ant-btn"
          >
            All
            <Icon
              type="down"
            />
          </Button>
        </Dropdown>,
        <span
          className="label"
        >
          Host filter:
        </span>,
        <Input
          autosize={false}
          className="value value-host-filter"
          disabled={false}
          onChange={[Function]}
          placeholder="Host filter"
          prefixCls="ant-input"
          type="text"
        />,
        <Table
          bordered={false}
          className="results-table"
          columns={
            Array [
              Object {
                "dataIndex": "host",
                "sorter": [Function],
                "title": "Host",
              },
              Object {
                "className": "column-count",
                "dataIndex": "count",
                "sorter": [Function],
                "title": "Count",
              },
              Object {
                "dataIndex": "latest",
                "render": [Function],
                "sorter": [Function],
                "title": "Latest",
              },
            ]
          }
          dataSource={
            Array [
              Object {
                "count": 10,
                "host": "two.com",
                "key": "two.com",
                "latest": 1509758701947.23,
              },
              Object {
                "count": 3,
                "host": "three.two.com",
                "key": "three.two.com",
                "latest": 1509758701932.883,
              },
              Object {
                "count": 1,
                "host": "four.three.two.com",
                "key": "four.three.two.com",
                "latest": 1509758701941.5847,
              },
            ]
          }
          indentSize={20}
          loading={false}
          locale={Object {}}
          pagination={false}
          prefixCls="ant-table"
          rowKey="key"
          rowSelection={null}
          showHeader={true}
          size="middle"
          useFixedHeader={false}
        />,
      ],
      "className": "dns-lookup",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "DNS lookups",
        },
        "ref": null,
        "rendered": "DNS lookups",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Host levels: ",
          "className": "label",
        },
        "ref": null,
        "rendered": "Host levels: ",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <Button
            className="value"
            clicked={false}
            ghost={false}
            loading={false}
            prefixCls="ant-btn"
          >
            All
            <Icon
              type="down"
            />
          </Button>,
          "mouseEnterDelay": 0.15,
          "mouseLeaveDelay": 0.1,
          "overlay": <Menu
            className=""
            onClick={[Function]}
            prefixCls="ant-menu"
            theme="light"
          >
            <MenuItem
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onSelect={[Function]}
            >
              All
            </MenuItem>
            <MenuItem
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onSelect={[Function]}
            >
              1
            </MenuItem>
            <MenuItem
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onSelect={[Function]}
            >
              2
            </MenuItem>
            <MenuItem
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onSelect={[Function]}
            >
              3
            </MenuItem>
            <MenuItem
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onSelect={[Function]}
            >
              4
            </MenuItem>
          </Menu>,
          "placement": "bottomLeft",
          "prefixCls": "ant-dropdown",
          "trigger": Array [
            "click",
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "All",
              <Icon
                type="down"
              />,
            ],
            "className": "value",
            "clicked": false,
            "ghost": false,
            "loading": false,
            "prefixCls": "ant-btn",
          },
          "ref": null,
          "rendered": Array [
            "All",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "type": "down",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Host filter:",
          "className": "label",
        },
        "ref": null,
        "rendered": "Host filter:",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "autosize": false,
          "className": "value value-host-filter",
          "disabled": false,
          "onChange": [Function],
          "placeholder": "Host filter",
          "prefixCls": "ant-input",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "bordered": false,
          "className": "results-table",
          "columns": Array [
            Object {
              "dataIndex": "host",
              "sorter": [Function],
              "title": "Host",
            },
            Object {
              "className": "column-count",
              "dataIndex": "count",
              "sorter": [Function],
              "title": "Count",
            },
            Object {
              "dataIndex": "latest",
              "render": [Function],
              "sorter": [Function],
              "title": "Latest",
            },
          ],
          "dataSource": Array [
            Object {
              "count": 10,
              "host": "two.com",
              "key": "two.com",
              "latest": 1509758701947.23,
            },
            Object {
              "count": 3,
              "host": "three.two.com",
              "key": "three.two.com",
              "latest": 1509758701932.883,
            },
            Object {
              "count": 1,
              "host": "four.three.two.com",
              "key": "four.three.two.com",
              "latest": 1509758701941.5847,
            },
          ],
          "indentSize": 20,
          "loading": false,
          "locale": Object {},
          "pagination": false,
          "prefixCls": "ant-table",
          "rowKey": "key",
          "rowSelection": null,
          "showHeader": true,
          "size": "middle",
          "useFixedHeader": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            DNS lookups
          </h2>,
          <span
            className="label"
          >
            Host levels: 
          </span>,
          <Dropdown
            mouseEnterDelay={0.15}
            mouseLeaveDelay={0.1}
            overlay={
              <Menu
                className=""
                onClick={[Function]}
                prefixCls="ant-menu"
                theme="light"
              >
                <MenuItem
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onSelect={[Function]}
                >
                  All
                </MenuItem>
                <MenuItem
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onSelect={[Function]}
                >
                  1
                </MenuItem>
                <MenuItem
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onSelect={[Function]}
                >
                  2
                </MenuItem>
                <MenuItem
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onSelect={[Function]}
                >
                  3
                </MenuItem>
                <MenuItem
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onSelect={[Function]}
                >
                  4
                </MenuItem>
              </Menu>
            }
            placement="bottomLeft"
            prefixCls="ant-dropdown"
            trigger={
              Array [
                "click",
              ]
            }
          >
            <Button
              className="value"
              clicked={false}
              ghost={false}
              loading={false}
              prefixCls="ant-btn"
            >
              All
              <Icon
                type="down"
              />
            </Button>
          </Dropdown>,
          <span
            className="label"
          >
            Host filter:
          </span>,
          <Input
            autosize={false}
            className="value value-host-filter"
            disabled={false}
            onChange={[Function]}
            placeholder="Host filter"
            prefixCls="ant-input"
            type="text"
          />,
          <Table
            bordered={false}
            className="results-table"
            columns={
              Array [
                Object {
                  "dataIndex": "host",
                  "sorter": [Function],
                  "title": "Host",
                },
                Object {
                  "className": "column-count",
                  "dataIndex": "count",
                  "sorter": [Function],
                  "title": "Count",
                },
                Object {
                  "dataIndex": "latest",
                  "render": [Function],
                  "sorter": [Function],
                  "title": "Latest",
                },
              ]
            }
            dataSource={
              Array [
                Object {
                  "count": 10,
                  "host": "two.com",
                  "key": "two.com",
                  "latest": 1509758701947.23,
                },
                Object {
                  "count": 3,
                  "host": "three.two.com",
                  "key": "three.two.com",
                  "latest": 1509758701932.883,
                },
                Object {
                  "count": 1,
                  "host": "four.three.two.com",
                  "key": "four.three.two.com",
                  "latest": 1509758701941.5847,
                },
              ]
            }
            indentSize={20}
            loading={false}
            locale={Object {}}
            pagination={false}
            prefixCls="ant-table"
            rowKey="key"
            rowSelection={null}
            showHeader={true}
            size="middle"
            useFixedHeader={false}
          />,
        ],
        "className": "dns-lookup",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "DNS lookups",
          },
          "ref": null,
          "rendered": "DNS lookups",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Host levels: ",
            "className": "label",
          },
          "ref": null,
          "rendered": "Host levels: ",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <Button
              className="value"
              clicked={false}
              ghost={false}
              loading={false}
              prefixCls="ant-btn"
            >
              All
              <Icon
                type="down"
              />
            </Button>,
            "mouseEnterDelay": 0.15,
            "mouseLeaveDelay": 0.1,
            "overlay": <Menu
              className=""
              onClick={[Function]}
              prefixCls="ant-menu"
              theme="light"
            >
              <MenuItem
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onSelect={[Function]}
              >
                All
              </MenuItem>
              <MenuItem
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onSelect={[Function]}
              >
                1
              </MenuItem>
              <MenuItem
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onSelect={[Function]}
              >
                2
              </MenuItem>
              <MenuItem
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onSelect={[Function]}
              >
                3
              </MenuItem>
              <MenuItem
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onSelect={[Function]}
              >
                4
              </MenuItem>
            </Menu>,
            "placement": "bottomLeft",
            "prefixCls": "ant-dropdown",
            "trigger": Array [
              "click",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "All",
                <Icon
                  type="down"
                />,
              ],
              "className": "value",
              "clicked": false,
              "ghost": false,
              "loading": false,
              "prefixCls": "ant-btn",
            },
            "ref": null,
            "rendered": Array [
              "All",
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "type": "down",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Host filter:",
            "className": "label",
          },
          "ref": null,
          "rendered": "Host filter:",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "autosize": false,
            "className": "value value-host-filter",
            "disabled": false,
            "onChange": [Function],
            "placeholder": "Host filter",
            "prefixCls": "ant-input",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "bordered": false,
            "className": "results-table",
            "columns": Array [
              Object {
                "dataIndex": "host",
                "sorter": [Function],
                "title": "Host",
              },
              Object {
                "className": "column-count",
                "dataIndex": "count",
                "sorter": [Function],
                "title": "Count",
              },
              Object {
                "dataIndex": "latest",
                "render": [Function],
                "sorter": [Function],
                "title": "Latest",
              },
            ],
            "dataSource": Array [
              Object {
                "count": 10,
                "host": "two.com",
                "key": "two.com",
                "latest": 1509758701947.23,
              },
              Object {
                "count": 3,
                "host": "three.two.com",
                "key": "three.two.com",
                "latest": 1509758701932.883,
              },
              Object {
                "count": 1,
                "host": "four.three.two.com",
                "key": "four.three.two.com",
                "latest": 1509758701941.5847,
              },
            ],
            "indentSize": 20,
            "loading": false,
            "locale": Object {},
            "pagination": false,
            "prefixCls": "ant-table",
            "rowKey": "key",
            "rowSelection": null,
            "showHeader": true,
            "size": "middle",
            "useFixedHeader": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
